components/common/feeds/AuthorFeed.tsx

```diff
@@ -4,6 +4,66 @@ import { Posts } from '../posts/Posts'
 import client from '@/lib/client'
 import { useState } from 'react'
 import { useRepoAndProfile } from '@/repositories/useRepoAndProfile'
+import { AppBskyFeedDefs } from '@atproto/api'
+import { TypeFilterKey, TypeFiltersByKey } from '../posts/constants'
+
+const getAuthorFeed = async ({ id, pageParam, typeFilter, options }) => {
+  const limit = 30
+  let filteredFeed: AppBskyFeedDefs.FeedViewPost[] = []
+  let cursor = pageParam
+  const isPostFilter = [
+    TypeFiltersByKey.posts_no_replies.key,
+    TypeFiltersByKey.posts_with_media.key,
+  ].includes(typeFilter)
+  const isQuoteOrRepostFilter = [
+    TypeFiltersByKey.reposts.key,
+    TypeFiltersByKey.quotes.key,
+    TypeFiltersByKey.quotes_and_reposts.key,
+  ].includes(typeFilter)
+
+  while (filteredFeed.length < limit) {
+    const { data } = await client.api.app.bsky.feed.getAuthorFeed(
+      {
+        limit,
+        actor: id,
+        cursor,
+        ...(isPostFilter ? { filter: typeFilter } : {}),
+      },
+      options,
+    )
+
+    // Only repost/quote post filters are applied on the client side
+    if (!isQuoteOrRepostFilter) {
+      return data
+    }
+
+    const newFilteredItems = data.feed.filter((item) => {
+      const isRepost = item.reason?.$type === 'app.bsky.feed.defs#reasonRepost'
+      const isQuotePost =
+        item.post.embed?.$type === 'app.bsky.embed.record#view'
+      if (typeFilter === TypeFiltersByKey.reposts.key) {
+        return isRepost
+      }
+      if (typeFilter === TypeFiltersByKey.quotes.key) {
+        // When a quoted post is reposted, we don't want to consider that a quote post
+        return isQuotePost && !isRepost
+      }
+      return isRepost || isQuotePost
+    })
+
+    filteredFeed = [...filteredFeed, ...newFilteredItems]
+
+    // If no more items are available, break the loop to prevent infinite requests
+    if (!data.cursor) {
+      break
+    }
```

Comment:
Truly no real issue here, but I often find myself reaching for this pattern when paginating which handles this control flow:
```ts
let cursor: string | undefined
do {
  const page = await getPage(cursor)
  cursor = page.cursor
} while (cursor)
```

Reply:
this is a tiny bit different though, we don't want to iterate through all pages, we only want to iterate until the original page size is reached but before that, if we go through all pages already, we want to break even if the page size is not reached. we could combine the conditions but it might be a bit more confusing that way.